### DISCUSSION TITLE
Getting "This test run was triggered on the branch master, while semantic-release is configured to only publish from main" in cloudbuild

### DISCUSSION THREAD

**Discussion by gabriellamichelle** [semantic-release/semantic-release#3496] [Configuration]
Running `npx semantic-release --no-ci` inside a cloudbuild script that is triggered by a push to branch `main`.

.releaserc
```
{
  "branches": [
    "main", "master"
  ]
}
```

For some reason it is assuming that it is run on master branch even though the `BRANCH_NAME` and `REF` under cloudbuild is correct as `main`; and thus throwing the above error. I'm wondering what is missing.

**Comment by travi** [comment]
we would need more details from your build output in order to even attempt to help. i am not familiar with the details of cloudbuild, so the more details you can provide, the better

**Reply by travi** [reply]
for context, it looks like it was [considered](https://github.com/semantic-release/env-ci/pull/97) to add support for cloudbuild to [our env-ci library](https://github.com/semantic-release/env-ci) previously, but was [decided](https://github.com/semantic-release/env-ci/pull/97#issuecomment-540040921) that it did not provide the details in the way necessary to add the support to env-ci. you would likely need to overcome these shortcomings yourself if you choose to continue with the path of using codebuild for this
